WireAPI/Sources/WireAPI/Components/ResponseParser.swift

```diff
@@ -63,20 +82,26 @@ struct ResponseParser<Success> {
     }
 
     func parse(_ response: HTTPResponse) throws -> Success {
-        let code = response.code
-
-        guard let data = response.payload else {
-            throw ResponseParserError.missingPayload
+        guard !parseBlocks.isEmpty else {
+            throw ParsingError.noParseBlocksDefined
         }
-
+        
+        let code = response.code
+        let data = response.payload
+        
         for matcher in parseBlocks {
             if let success = try matcher(code, data) {
                 return success
             }
         }
-
-        let failure = try decoder.decode(FailureResponse.self, from: data)
-        throw failure
+        
+        if let data {
+            let failure = try decoder.decode(FailureResponse.self, from: data)
+            throw failure
+        } else {
+            throw ParsingError.nothingToParse
```

Comment:
I think rather than `nothingToParse` it would be more accurate to say the response had no parse result. Usually it should not happen but if it does then it would highlight to us that we have not handled the parsing correctly.

Reply:
makes sense thx